vidarr/wdl.py

```diff
@@ -207,6 +239,60 @@ def read_output(output: WDL.Decl):
         'workflow': doc.source_text,
         'accessoryFiles': {
             imported.uri: imported.doc.source_text for imported in doc.imports}}
+    
+    output_meta = doc.workflow.meta.get("output_meta", {})
+
+    output_lines = []
+
+    # Iterate over each output defined in the workflow
+    for output in doc.workflow.outputs:
+
+        output_line = str(output)
+
+        # Check if the output name exists in the output metadata
+        if output.name in output_meta:
+            output_metadata = output_meta[output.name]
+
+            if isinstance(output_metadata, dict) and 'vidarr_label' in output_metadata:
+                vidarr_label = output_metadata['vidarr_label']
+
+                if isinstance(output.type, WDL.Type.File) and not output.type.optional:
+
+                    # Construct the modified output line
+                    output_line = f"   Pair[File, Map[String, String]] {output.name} = ({output.expr}, {{\"vidarr_label\": \"{vidarr_label}\"}})"
+                    
+                elif isinstance(output.type, WDL.Type.Pair) and \
+                    isinstance(output.type.left_type, WDL.Type.File) and \
+                    isinstance(output.type.right_type, WDL.Type.Map):
+
+                    output_line = f"    Pair[File, Map[String, String]] {output.name} = {output.expr}"
+
+        # Append output lines to a consistently updated list
+        output_lines.append(output_line)
+
+    # Define the pattern for searching and replacing the output block
+    pattern = r"(?:workflow)([\s\S]*?)(?:output\s*{)([\s\S]*?)(?:\}\s*\n)"
+
+    # Search for the pattern in the workflow
+    match = re.search(pattern, workflow['workflow'], re.DOTALL)
+
+    # If the pattern is found, perform the replacement
+    if match:
+
+        # Extracting the part of the workflow string where the output block is located
+        output_block_start = match.start(2)
+        output_block_end = match.end(2)
+        output_block_text = match.group(2)
+
+        # Join all outputs
+        modified_output_block_text = "\n".join(output_lines)
```

Comment:
i think it might be this that's confusing the CI... the diff is just some spacing 
![image](https://github.com/oicr-gsi/vidarr-tools/assets/17747491/b811e5b0-7287-4c2b-9a4f-a17e820a6b00)
as long as the json is good and can be successfully POSTed to Vidarr then i'd say just update the test json

Reply:
Ya those tabs are only for vidarr_labels though, can I try POSTing to Vidarr or is Mike still testing?

Reply:
@mlaszloffy ?

Reply:
Please go ahead with vidarr-dev testing.